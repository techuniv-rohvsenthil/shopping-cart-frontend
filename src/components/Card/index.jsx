import React, {useState} from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';

const Card = (props) => {
  const {prodObj} = props;

  const [value, setValue] = useState(0);

  const increment = () => {
    if(value+1 < prodObj.prodQuantity){
      setValue(value+1);
    }
  }
  const decrement = () => {
    if(value-1 > 0){
      setValue(value-1);
    }
  }

  return (
    <div className={styles.card} >
        <img src={prodObj.prodImage} alt="prodImage" width="150px" height="150px"/>
        <div className={styles.details}>
          <div className={styles.text}>
            {prodObj.prodName}
          </div>
          <div className={styles.text}>
            MRP {prodObj.prodPrice} /-
          </div>
          <div className={styles.text}>
            <button data-testid="dec" type="button" className={styles.button} onClick={decrement}>-</button>
            {value}
            <button data-testid="inc" type="button" className={styles.button} onClick={increment}>+</button>
          </div>
        </div>
      </div>
  );
};


Card.propTypes = {
  prodObj: propTypes.objectOf(propTypes.string).isRequired,
};


export default Card;
