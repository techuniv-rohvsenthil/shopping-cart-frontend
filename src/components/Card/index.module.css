.card{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 20rem;
    border-radius: 5%;
    color: black;
    margin:1rem;
    font-family: 'Poppins';
    box-shadow: 0 3px 5px 2px rgba(105, 105, 105, .5);
}

.text{
    font-size: 1rem;
}

.button{
    margin: 1rem;
    color: black;
    width: 2rem;
    height: 2rem;
    background-color: #bdbdbd;
}

.sold{
    width: inherit;
    height: inherit;
    background-color: #bdbdbd;
    opacity: 0.8;
    color: maroon;
    font-size: 2rem;
    font-weight: 700;
    position: absolute;
    margin-top: 0.5rem;
}