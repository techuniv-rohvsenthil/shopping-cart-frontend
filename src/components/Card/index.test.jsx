import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Card from './index';

const prodObj = {
    prodName: "Apple - 1kg",
    prodPrice: '210',
    prodQuantity: '10',
    prodImage: 'https://techunic-eval4.s3.amazonaws.com/apple.jpg',
  }

describe('the Card component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Card  prodObj={prodObj}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  xit('should increment count when the + button is clicked', () => {
    const { getByTestId } = render(<Card prodObj={prodObj}/>);
    fireEvent.click(getByTestId('inc'));
    
  });

});
