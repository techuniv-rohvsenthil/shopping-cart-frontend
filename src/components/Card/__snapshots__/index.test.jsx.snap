// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Card component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="prodImage"
      height="150px"
      src="https://techunic-eval4.s3.amazonaws.com/apple.jpg"
      width="150px"
    />
    <div>
      <div>
        Apple - 1kg
      </div>
      <div>
        MRP 210 /-
      </div>
      <div>
        <button
          data-testid="dec"
          type="button"
        >
          -
        </button>
        0
        <button
          data-testid="inc"
          type="button"
        >
          +
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
